File: app/src/main/java/com/mikkipastel/blog/DAO/MikkiBlog.java

package com.mikkipastel.blog.DAO;

import java.util.ArrayList;
import java.util.List;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class MikkiBlog {

    @SerializedName("kind")
    @Expose
    private String kind;
    @SerializedName("items")
    @Expose
    private List<Item> items = new ArrayList<Item>();
    @SerializedName("etag")
    @Expose
    private String etag;

    /**
     * 
     * @return
     *     The kind
     */
    public String getKind() {
        return kind;
    }

    /**
     * 
     * @param kind
     *     The kind
     */
    public void setKind(String kind) {
        this.kind = kind;
    }

    /**
     * 
     * @return
     *     The items
     */
    public List<Item> getItems() {
        return items;
    }

    /**
     * 
     * @param items
     *     The items
     */
    public void setItems(List<Item> items) {
        this.items = items;
    }

    /**
     * 
     * @return
     *     The etag
     */
    public String getEtag() {
        return etag;
    }

    /**
     * 
     * @param etag
     *     The etag
     */
    public void setEtag(String etag) {
        this.etag = etag;
    }

}

File: app/src/main/java/com/mikkipastel/blog/model/MikkiBlog.kt
package com.mikkipastel.blog.model

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName

data class MikkiBlog(@SerializedName("kind") val kind: String?,
                     @SerializedName("nextPageToken") val nextPageToken: String?,
                     @SerializedName("items") val items: List<Item>,
                     @SerializedName("etag") val etag: String?): Parcelable {
    constructor(parcel: Parcel) : this(
            parcel.readString(),
            parcel.readString(),
            listOf<Item>().apply {
                parcel.readList(this, Item::class.java.classLoader)
            },
            parcel.readString())

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(kind)
        parcel.writeString(nextPageToken)
        parcel.writeList(items)
        parcel.writeString(etag)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<MikkiBlog> {
        override fun createFromParcel(parcel: Parcel): MikkiBlog {
            return MikkiBlog(parcel)
        }

        override fun newArray(size: Int): Array<MikkiBlog?> {
            return arrayOfNulls(size)
        }
    }
}